routing/osrm_path_segment_factory.hpp

```diff
@@ -0,0 +1,28 @@
+#pragma once
+
+#include "routing/loaded_path_segment.hpp"
+
+#include "std/unique_ptr.hpp"
+
+namespace routing
+{
+struct RoutingMapping;
+struct RawPathData;
+struct FeatureGraphNode;
+
+namespace turns
+{
+using TSeg = OsrmMappingTypes::FtSeg;
+
+// General constructor.
+unique_ptr<LoadedPathSegment> LoadedPathSegmentFactory(RoutingMapping & mapping,
```

Comment:
It's not a problem to create a no-arg ctor for LoadPathSegment. And processing of LoadPathSegment results will be almost the same as now.